sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -1222,19 +1222,33 @@ private static PartitionKeyInternal extractPartitionKeyValueFromDocument(
             InternalObjectNode document,
             PartitionKeyDefinition partitionKeyDefinition) {
         if (partitionKeyDefinition != null) {
-            String path = partitionKeyDefinition.getPaths().iterator().next();
-            List<String> parts = PathParser.getPathParts(path);
-            if (parts.size() >= 1) {
-                Object value = ModelBridgeInternal.getObjectByPathFromJsonSerializable(document, parts);
-                if (value == null || value.getClass() == ObjectNode.class) {
-                    value = ModelBridgeInternal.getNonePartitionKey(partitionKeyDefinition);
-                }
+            switch (partitionKeyDefinition.getKind()) {
+                case HASH:
+
+                String path = partitionKeyDefinition.getPaths().iterator().next();
+                List<String> parts = PathParser.getPathParts(path);
+                if (parts.size() >= 1) {
+                    Object value = ModelBridgeInternal.getObjectByPathFromJsonSerializable(document, parts);
+                    if (value == null || value.getClass() == ObjectNode.class) {
+                        value = ModelBridgeInternal.getNonePartitionKey(partitionKeyDefinition);
+                    }
 
-                if (value instanceof PartitionKeyInternal) {
-                    return (PartitionKeyInternal) value;
-                } else {
-                    return PartitionKeyInternal.fromObjectArray(Collections.singletonList(value), false);
+                    if (value instanceof PartitionKeyInternal) {
+                        return (PartitionKeyInternal) value;
+                    } else {
+                        return PartitionKeyInternal.fromObjectArray(Collections.singletonList(value), false);
+                    }
                 }
+                break;
+                case MULTI_HASH:
+                    Object[] partitionKeyValues = new Object[partitionKeyDefinition.getPaths().size()];
+                    for(int path_iter = 0 ; path_iter < partitionKeyDefinition.getPaths().size(); path_iter++)
```

Comment:
@moderakh any suggestions on elegant way of iteration.

Reply:
it is possible to use java stream.

partitionKeyDefinition.getPaths().stream().map .... 

but not much difference functional wise.